subprojects/native/src/main/java/org/gradle/internal/nativeintegration/services/NativeServices.java

```diff
@@ -109,7 +113,16 @@ public static synchronized void initialize(File userHomeDir, boolean initializeJ
                             throw ex;
                         }
                     }
-                    if (initializeJansi) {
+                    if (useNativeIntegrations && initializeAdditionalNativeLibraries) {
+                        useFileSystemWatching = true;
+                        try {
+                            FileEvents.init(nativeBaseDir);
+                        } catch (NativeIntegrationUnavailableException ex) {
+                            LOGGER.info("Native file system watching is not available for this operating system.", ex);
+                            useFileSystemWatching = false;
+                        }
+                    }
+                    if (initializeAdditionalNativeLibraries) {
```

Comment:
I think things would be more readable if this `if` was wrapping the one currently right above it.

Reply:
Done.